Automattic/jetpack
# Social: Add Threads connection UI

## Proposed changes:

* Update Threads icon color
* Add Threads to supported services for social connections
* Add badges support for services list in the modal
* Add "New" badge to Threads in the services list

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Sandbox D152810-code and follow the instructions given there
* Point your Jetpack site and `public-api.wordpress.com` to your sandbox
* Goto Social/Jetpack settings
* Open the connections modal
* Confirm that you see Threads in the services list
* Confirm that you see the "New" badge added to it
* Confirm that you see the correct logo and its color
* Connect a threads account
* Confirm that you see the Threads connection in the list
* Confirm that you also see it in the modal
* Confirm that you see the added connection in Calypso
* Goto post editor
* Confirm that you see the Threads connection there, along with its icon
* Share a post to Threads
* Confirm that it works

<img width="1050" alt="image" src="https://github.com/Automattic/jetpack/assets/18226415/d66013ee-9b81-47bc-a6b9-78693e871fc5">

<img width="1074" alt="image" src="https://github.com/Automattic/jetpack/assets/18226415/4d1fa5b1-da75-4c8e-83d5-863b0e626f79">

<img width="1048" alt="Screenshot 2024-06-21 at 12 22 42 PM" src="https://github.com/Automattic/jetpack/assets/18226415/f937be37-a147-4be3-9089-f1b94375698a">
<img width="643" alt="Screenshot 2024-06-21 at 12 22 49 PM" src="https://github.com/Automattic/jetpack/assets/18226415/3db3b3b4-a792-4190-9fbe-828ffaf70303">
<img width="641" alt="Screenshot 2024-06-21 at 12 23 00 PM" src="https://github.com/Automattic/jetpack/assets/18226415/1b47d3a9-53a8-48f1-90fc-a68b8f60829a">
<img width="1051" alt="Screenshot 2024-06-21 at 12 23 13 PM" src="https://github.com/Automattic/jetpack/assets/18226415/11d7ac27-82a9-40ad-ad99-12f935614aa5">
<img width="291" alt="Screenshot 2024-06-21 at 12 23 30 PM" src="https://github.com/Automattic/jetpack/assets/18226415/1eb5eaa1-3566-45b3-9a5b-0831dc91c6e6">
